e2e/test/csi_driver_test.go

```diff
@@ -61,33 +61,71 @@ var _ = Describe("Linode CSI Driver", func() {
 
 	Describe("Test", func() {
 		Context("Simple", func() {
+			var (
+				f    *framework.Invocation
+				size string
+				// storageClass = "linode-block-storage"
+				sts          *appsv1.StatefulSet
+				pod          *core.Pod
+				pvc          *core.PersistentVolumeClaim
+				file         = "/data/file.txt"
+				storageClass = "linode-block-storage"
+
+				writeFile = func(filename string) {
+					By("Writing a file into the Pod")
+					err := f.WriteFileIntoPod(filename, pod)
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+				readFile = func(filename string) {
+					By("Checking if the created file is in the Pod")
+					err := f.CheckIfFileIsInPod(filename, pod)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			)
 			Context("Block Storage", func() {
 				JustBeforeEach(func() {
 					By("Creating Persistent Volume Claim")
-					pvc = f.GetPersistentVolumeClaimObject(size, f.StorageClass, false)
-					err = f.CreatePersistentVolumeClaim(pvc)
+					pvc = f.GetPersistentVolumeClaimObject("test-pvc", f.Namespace(), size, storageClass, false)
+					err := f.CreatePersistentVolumeClaim(pvc)
 					Expect(err).NotTo(HaveOccurred())
 
 					By("Creating Pod with PVC")
-					pod = f.GetPodObject(podName1, pvc.Name)
+					pod = framework.GetPodObject("test-pod", f.Namespace(), pvc.Name)
 					err = f.CreatePod(pod)
 					Expect(err).NotTo(HaveOccurred())
 				})
 
 				AfterEach(func() {
 					By("Deleting the Pod with PVC")
-					err = f.DeletePod(pod.Name)
+					err := f.DeletePod(pod.ObjectMeta)
 					Expect(err).NotTo(HaveOccurred())
 
 					By("Waiting for the Volume to be Detached")
-					waitForOperation()
+					volumeID, err := f.GetVolumeID(pvc)
+
+					Eventually(func() bool {
+						isAttached, err := f.IsVolumeDetached(volumeID)
```

Comment:
`isAttached, _ = f.IsVolumeDetached`?

Reply:
oh ya that's cleaner lemme do that

Reply:
ah but to your other point, we could dump these errors for debugging purposes?

Reply:
I'm not saying to discard the error, I was pointing to the fact that the variable is named `isAttached` but the method presumably returns the opposite.